### PR TITLE
Add --system and --all-logs support to bosh logs --director

### PR DESCRIPTION
**Pull request by jpalermo**
Follow on to #610 and #614

### CODE REVIEW

**Review by klakin-pivotal — CHANGES_REQUESTED** [review]

File: director/deployment_test.go [hunk 1]
```diff
@@ -239,99 +239,7 @@ var _ = Describe("Deployment", func() {
 			)
 
 			result, err := deployment.FetchLogs(
-				NewAllOrInstanceGroupOrInstanceSlug("job", "id"), []string{"f1", "f2"}, true, false, false)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(result).To(Equal(LogsResult{BlobstoreID: "logs-blob-id", SHA1: ""}))
-		})
-
-		It("is correctly fetches job, agent, and system logs when --all is passed", func() {
```
[the remark this answers is not in the document]
Comment (jpalermo) [reply director/deployment_test.go:247]: The code in deployment no longer has this logic though. The log type is passed in as a string and just directly used. Previously the logic for determining which string to send was in the deployment, so the tests made sense here. The deleted tests are all identical to the [test that was left](https://github.com/cloudfoundry/bosh-cli/pull/616/files#diff-1c255e0d31c5bb8ffb4126a80bfe5620c98958bad4884cc8e35435bd7fe429fbR224) except for the value of the logType, and you can put whatever string value you want in there.

Now the log cmd determines what the log type should be (and the logic is shared between the two different log commands) and tests that verify it passes the correct value down to the deployment are [here](https://github.com/cloudfoundry/bosh-cli/pull/616/files#diff-22e518acf530a844ab4051bf614550b8ebfca84040fb558fab1c4fc3c2b0fed8R144), [here](https://github.com/cloudfoundry/bosh-cli/pull/616/files#diff-22e518acf530a844ab4051bf614550b8ebfca84040fb558fab1c4fc3c2b0fed8R127), [here](https://github.com/cloudfoundry/bosh-cli/pull/616/files#diff-22e518acf530a844ab4051bf614550b8ebfca84040fb558fab1c4fc3c2b0fed8R110), and [here](https://github.com/cloudfoundry/bosh-cli/pull/616/files#diff-22e518acf530a844ab4051bf614550b8ebfca84040fb558fab1c4fc3c2b0fed8R85)

[the remark this answers is not in the document]
Comment (klakin-pivotal) [reply director/deployment_test.go:247]: I like it when the tests document the intent of the code.

So it'd be good if the tests document that either
*  when you put any and all combinations of expected values in, you get the expected values out
* _anything_ you feed to this function gets plopped into the expected place in the URI that it returns and uses in that `get` request

And if we're doing the latter, the data we feed in in the test should probably be data that requires URI-escaping... just to make sure that that part of the code works as it should.

[the remark this answers is not in the document]
Comment (jpalermo) [reply director/deployment_test.go:247]: 👍 I updated the test and assertion to include a URI escaped example

[the remark this answers is not in the document]
Comment (klakin-pivotal) [reply director/deployment_test.go:247]: Love it.

**Review by klakin-pivotal — APPROVED** [review]

**Pull request merged**